onnxruntime/core/providers/cpu/tensor/gather_elements.cc

```diff
@@ -173,17 +159,24 @@ static void core_impl(const Tensor* input_tensor, const Tensor* indices_tensor,
       base_offset = compute_base_offset(process_dims, input_shape_pitches, axis);
 
       // process 1 chunk of 'inner dimension' length
-      for (int64_t i = 0; i < inner_dim_size; ++i) {
-        // optimizer will remove the redundant if/else block based on 'is_string' template parameter
-        if (is_string) {
-          output_data[++output_counter] = input_data[base_offset + (indices_data[++indices_counter] * input_shape_pitches[axis]) + i];
-        } else {
+      if (is_string) {
+        for (int64_t i = 0; i < inner_dim_size; ++i) {
+          // optimizer will remove the redundant if/else block based on 'is_string' template parameter
```

Comment:
nit: should the comment be kept closer to the if...else (now that if and for are inter-changed). ? Same for the other 2 instances below...